projects/ADuCM3029_demo_cn0428_cn0429/src/ADuCM3029_demo_cn0428_cn0429.cpp

```diff
@@ -0,0 +1,1307 @@
+						     "Config of sensor on site 3:%s", _EOS);
+	if (sensor_address == 0x0D) sprintf ((char*)&TXbuff,
+						     "Config of sensor on site 4:%s", _EOS);
+	UART_TX((const char*)TXbuff);
+	delay_ms(5);
+
+	flushBuff(pBuff, sizeof(pBuff));
+	eSensorResult = pGasSensor->ReadTIAGain(pBuff);
+	if(eSensorResult != SENSOR_ERROR_NONE) return eSensorResult;
+	flushBuff(TXbuff, sizeof(TXbuff));
+	sprintf((char*)TXbuff,"RTIA = %s ohm%s",rtia[pBuff[0]], _EOS);
+	UART_TX((const char*)TXbuff);
+	delay_ms(5);
+
+	flushBuff(pBuff, sizeof(pBuff));
+	eSensorResult = pGasSensor->ReadRload(pBuff);
+	if(eSensorResult != SENSOR_ERROR_NONE) return eSensorResult;
+	flushBuff(TXbuff, sizeof(TXbuff));
+	sprintf((char*)TXbuff,"Rload = %s ohm%s",rload[pBuff[0]], _EOS);
+	UART_TX((const char*)TXbuff);
+	delay_ms(5);
+
+	eSensorResult = pGasSensor->ReadSensorBias(&sigVal);
+	if(eSensorResult != SENSOR_ERROR_NONE) return eSensorResult;
+	flushBuff(TXbuff, sizeof(TXbuff));
+	sprintf((char*)TXbuff,"Vbias = %d mV%s",sigVal, _EOS);
+	UART_TX((const char*)TXbuff);
+	delay_ms(5);
+
+	eSensorResult = pGasSensor->ReadSensorSensitivity(&unsiglVal);
+	if(eSensorResult != SENSOR_ERROR_NONE) return eSensorResult;
+	flushBuff(TXbuff, sizeof(TXbuff));
+	sprintf((char*)TXbuff,"Sensitivity = %.2f nA/ppm%s", (unsiglVal / 100.0), _EOS);
+	UART_TX((const char*)TXbuff);
+	delay_ms(5);
+
+	eSensorResult = pGasSensor->ReadMeasurementTime(&unsigVal);
+	if(eSensorResult != SENSOR_ERROR_NONE) return eSensorResult;
+	flushBuff(TXbuff, sizeof(TXbuff));
+	sprintf((char*)TXbuff,"Measurement Time = %d msec%s", unsigVal, _EOS);
+	UART_TX((const char*)TXbuff);
+	delay_ms(5);
+
+	// === NOT USED IN THIS VERSION! ===
+	/*flushBuff(pBuff, sizeof(pBuff));
+	eSensorResult = pGasSensor->ReadTempCompCfg(pBuff);
+	if(eSensorResult != SENSOR_ERROR_NONE) return eSensorResult;
+	flushBuff(TXbuff, sizeof(TXbuff));
+	if (pBuff[0] == 0) sprintf((char*)TXbuff,"Temp Comp = disabled%s", _EOS);
+	if (pBuff[0] == 1) sprintf((char*)TXbuff,"Temp Comp = enabled%s", _EOS);
+	UART_TX((const char*)TXbuff);
+	delay_ms(5);*/
+
+	eSensorResult = pGasSensor->close();
+	delay_ms(5);
+
+	return eSensorResult;
+}
+
+/*!
+ * @brief      Reads averaged ppb value from gas sensor
+ *
+ * @details    Reads sensor data utilizing custom gas sensor I2C library and sends this data over UART.
+ */
+int32_t SensorReadoutPPB(uint8_t sensor_address)
+{
+	int32_t PPB_Gas_Reading = 0;
+	eSensorResult = pGasSensor->openWithAddr(sensor_address);
+	eSensorResult = pGasSensor->ReadDataPPB(&PPB_Gas_Reading);
+	eSensorResult = pGasSensor->close();
+	delay_ms(10);
+	return PPB_Gas_Reading;
+}
+
+int32_t ExtractCfgVal()
+{
+	uint8_t buffer[16] = {0};
+	uint8_t read_char;
+	int i = 0;
+	for (i = 0; i < 16; i++) {
+		read_char = GetChar();
+		if (read_char == '\0') break;
+		else buffer[i] = read_char;
+	}
+	DeleteChar(i + 1);
+	return strtol((const char*)buffer, NULL, 10);
+}
+
+uint8_t *int2binString(int32_t a, uint8_t *buff, uint8_t bufSize)
```

Comment:
i think you can use function itoa() for this.

Reply:
itoa is not part of C++, right?
"Portability:
This function is not defined in ANSI-C and is not part of C++, but is supported by some compilers." - http://www.cplusplus.com/reference/cstdlib/itoa/

Reply:
You are right. Thanks.